packages/brain/src/modules/cron/trackValidatorsPerformance.ts

```diff
@@ -0,0 +1,167 @@
+ * finalized changes, the issue will be logged and stored in the DB.
+ *
+ * @param validatorPubkeys - The pubkeys of the validators to track.
+ * @param postgresClient - Postgres client to interact with the DB.
+ * @param beaconchainApi - Beaconchain API client to interact with the Beaconchain API.
+ * @param minGenesisTime - The minimum genesis time of the chain.
+ */
+export async function trackValidatorsPerformance({
+  brainDb,
+  postgresClient,
+  beaconchainApi,
+  minGenesisTime,
+  secondsPerSlot
+}: {
+  brainDb: BrainDataBase;
+  postgresClient: PostgresClient;
+  beaconchainApi: BeaconchainApi;
+  minGenesisTime: number;
+  secondsPerSlot: number;
+}): Promise<void> {
+  // get validator pubkeys
+  const validatorPubkeys = Object.keys(brainDb.getData());
+
+  // skip if no validator indexes
+  if (validatorPubkeys.length === 0) return;
+
+  const epochFinalized = await beaconchainApi.getEpochHeader({ blockId: "finalized" });
+  let newEpochFinalized = epochFinalized;
+
+  while (epochFinalized === newEpochFinalized) {
+    try {
+      logger.debug(`${logPrefix}Epoch finalized: ${epochFinalized}`);
+
+      // get validator indexes
+      // TODO: could be a better way to get the indexes? i.e store them in cache or brain DB
+      logger.debug(`${logPrefix}Getting validator indexes from pubkeys`);
+      const validatorIndexes = await Promise.all(
+        validatorPubkeys.map(
+          async (pubkey) => (await beaconchainApi.getStateValidator({ state: "finalized", pubkey })).data.index
+        )
+      );
+      logger.debug(`${logPrefix}Validator indexes: ${validatorIndexes}`);
+
+      // get only active validators
+      logger.debug(`${logPrefix}$Getting active validators`);
+      const activeValidators = (
+        await beaconchainApi.postStateValidators({
+          body: {
+            ids: validatorIndexes,
+            statuses: validatorIndexes.map(() => ValidatorStatus.ACTIVE_ONGOING)
+          },
+          stateId: "finalized"
+        })
+      ).data.map((validator) => validator.index);
+      logger.debug(`${logPrefix}Active validators: ${activeValidators}`);
+
+      if (activeValidators.length === 0) {
+        logger.info(`${logPrefix}No active validators found`);
+        return;
+      }
+
+      // check node health
+      logger.debug(`${logPrefix}Checking node health`);
+      const { el_offline, is_syncing } = (await beaconchainApi.getSyncingStatus()).data;
+      logger.debug(`${logPrefix}EL Node offline: ${el_offline}, Node syncing: ${is_syncing}`);
+      if (el_offline) throw Error("EL Node is offline");
+      if (is_syncing) throw Error("Node is syncing");
+
+      // get block attestations
+      logger.debug(`${logPrefix}Getting attestations rewards for epoch ${epochFinalized}`);
+      const validatorsAttestationsRewards = (
+        await beaconchainApi.getAttestationsRewards({
```

Comment:
Im worried about requesting the consensus client 100s of times if the brain has +100 keys loaded. Could take a lot of time, maybe more than a minute & epoch isnt abailable anymore